### ISSUE TITLE
No line number generated when an error occurs in isAfter

### ISSUE THREAD

**Issue by adrianffletcher** [accordproject/ergo#622]
The following Ergo code results
```
  clause generateIsAfterError(request : TestRequest) : MyResponse {
    let a : DateTime? = some(now());
    let b : DateTime = now();
    enforce isAfter(a,b) else
      throw ErgoErrorResponse{ message : "Time is going backwards!" };
    return MyResponse{ output: "Test Complete" }
  }
```
Results in this error message:
```
Adrians-MBP:helloconcept adrianfletcher$ cicero execute
7:47:27 PM - info: Using current directory as template folder
7:47:27 PM - info: Loading a default sample.txt file.
7:47:27 PM - info: Loading a default request.json file.
7:47:28 PM - error: Type error. This operator received unexpected arguments of type `DateTime?'  and `DateTime'.
```

It would be great to get the line number that the error occurred at.
[helloconcept@0.1.1.zip](https://github.com/accordproject/ergo/files/3311725/helloconcept%400.1.1.zip)

**Comment by jeromesimeon** [comment]
Thanks for the report. This is most likely not specific to `isAfter` but a not-so-great error message for operators called from within the standard library.

**Comment by jeromesimeon** [comment]
The table used to find built-in operators (which `isAfter` is one of) was using dummy locations:

https://github.com/accordproject/ergo/blob/1786e88e649c24aec81ec301d72f873b28d4e236/mechanization/ErgoC/Lang/ErgoCStdlib.v#L241

An on-going fix looks promising:
```
bash-3.2$ ~/git/cicero/packages/cicero-cli/index.js parse 
16:07:28 - info: Using current directory as template folder
16:07:28 - info: Loading a default sample.txt file.
16:07:36 - error: Type error (at file lib/logic.ergo line 36 col 12). This operator received unexpected arguments of type `DateTime?'  and `DateTime'.
    enforce isAfter(a,b) else
            ^^^^^^^^^^^^     
```

**Comment by jeromesimeon** [comment]
Fixed in #624
